packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_transaction_wrappers.dart

```diff
@@ -188,8 +196,8 @@ class SKPaymentTransactionWrapper {
   String toString() => _$SKPaymentTransactionWrapperToJson(this).toString();
 
   /// The payload that is used to finish this transaction.
-  Map<String, String> toFinishMap() => {
+  Map<String, String?> toFinishMap() => <String, String?>{
         "transactionIdentifier": this.transactionIdentifier,
```

Comment:
@stuartmorgan I updated here so when `transactionIdentifier` is null, the map contains null. Because transactionIdentifier is nullable, the platform code is expecting a null value from the map.
I also added tests for this.